### ISSUE TITLE
Payment transaction issue

### ISSUE THREAD

**Issue by gagan50** [PaystackHQ/paystack-android#67]
String cardNumber = "5060 6666 6666 6666 666";
        int expiryMonth = 10; //any month in the future
        int expiryYear = 22; // any year in the future. '2018' would work also!
        String cvv = "123";  // cvv of the test card

Error! Please try again

**Comment by Ikhiloya** [comment]
What exactly is the error? Could you share your logs

**Comment by gagan50** [comment]
network calls mean?

**Comment by gagan50** [comment]
compile 'co.paystack.android:paystack:3.0.10'

**Comment by Ikhiloya** [comment]
Like Http Requests

**Comment by gagan50** [comment]
String cardNumber = "5060 6666 6666 6666 666";
        int expiryMonth = 10; //any month in the future
        int expiryYear = 22; // any year in the future. '2018' would work also!
        String cvv = "123";  // cvv of the test card

        card = new Card(cardNumber, expiryMonth, expiryYear, cvv);
        if (card.isValid()) {
            performCharge();
            // charge card
        } else {
            //do something
        }


  public void performCharge() {
        //create a Charge object
        Charge charge = new Charge();
        charge.setCard(card); //sets the card to charge
        charge.setAmount(10);
        charge.setReference("ChargedFromAndroid_" + Calendar.getInstance().getTimeInMillis());
        charge.setEmail("gagan@mailinator.com");
        PaystackSdk.chargeCard(LoginActivity.this, charge, new Paystack.TransactionCallback() {
            @Override
            public void onSuccess(Transaction transaction) {
                Alerts.showLog("transactions", ">>>>" + transaction.toString());
                // This is called only after transaction is deemed successful.
                // Retrieve the transaction, and send its reference to your server
                // for verification.
            }

            @Override
            public void beforeValidate(Transaction transaction) {
                Alerts.showLog("transactions", ">beforeValidate>>>" + transaction.toString());

                // This is called only before requesting OTP.
                // Save reference so you may send to server. If
                // error occurs with OTP, you should still verify on server.
            }

            @Override
            public void onError(Throwable error, Transaction transaction) {
                //handle error here
                Alerts.showLog("transactions", ">error>>>" + error.getMessage());

            }

        });
    }


that my code i have implemented

**Comment by gagan50** [comment]
on transaction sucess how i can get detail of transaction

**Comment by Ikhiloya** [comment]
You can get it from the Transaction object onSuccess

**Comment by gagan50** [comment]
can i get transaction id? or anything else

**Comment by Ikhiloya** [comment]
The Transaction object contains some fields. Check the Transaction object. You'd get a transaction reference

**Comment by gagan50** [comment]
thanks i got it

**Comment by Ikhiloya** [comment]
You're welcome
!

**Comment by Ikhiloya** [comment]
please close this issue @ibrahimlawal
